package/MDAnalysis/core/topologyobjects.py

```diff
@@ -539,18 +539,19 @@ def __init__(self, bondidx, universe, btype=None, type=None, guessed=None,
             raise ValueError("Unsupported btype, use one of '{}'"
                              "".format(', '.join(_BTYPE_TO_SHAPE)))
 
+        bondidx = np.asarray(bondidx)
         nbonds = len(bondidx)
         # remove duplicate bonds
         if type is None:
-            type = np.repeat(None, nbonds).reshape(nbonds, 1)
```

Comment:
If this is the fix for #2392 then I would separate it from this PR and make it a new one. It seems extra effort but it's much easier to discuss and review small things, especially if one is simple and the other complex.

Reply:
I created #2403 . Should the changes be removed from this PR? I think this PR was failing tests without it (how the bug was discovered in the first place)

Reply:
In principle, yes, remove. In practice, wait until #2403 is merged. Given that this PR will likely get squash-merged anyway, you can wait if any conflicts emerge.